### PR TITLE
W-8901531 - OFM installer optionally installs OFM Experience

### PR DESCRIPTION
**Pull request by seanpat09**
For testing:
* Create a scratch org that doesn't have Digital Experiences enabled and launch staging installer for this org
   * Installer should display warning that you need to have Digital Experiences enabled to install OFM Experience Template
   * Option to install OFM Experience Template should not be visible

* Create a scratch org that has Digital Experiences enabled and launch staging installer for this org
   * Installer should not display warning that you need to have Digital Experiences enabled to install OFM Experience Template
   * Option to install OFM Experience Template should be visible and selected by default
   * When option is selected OFM Experience Template package should be installed
   * When option is NOT selected OFM Experience Template package should NOT be installed

Adds optional step to install the OFM Experience package if Digital Experiences is enabled in the target org. If Digital Experiences is not enabled, the option is hidden in the installer. When the OFM Experience package gets released, we'll need to swap out `install_managed_beta` with `install_managed`

# Critical Changes

# Changes
- If Digital Experiences is enabled, the Outbound Funds Module will now install the Outbound Funds Module Experience Builder package. 
# Issues Closed

# New Metadata

# Deleted Metadata

# Definition of Done
Refer to [Definition of Done](https://salesforce.quip.com/9P7hAOPHJJyU) to see any additional details for the items below:
- [ ] ~Any net new LWC work has JEST test coverage 50% or above~
- [ ] ~Default Sa11y tests pass for all LWC components~
- [ ] ~🔒 Secure both Front-end (LWC) & back-end (Apex) as necessary~
- [ ] ~🔑 Grant users access in Permission Sets (Object, Field, Apex Class) as necessary~
- [x] Link the pull request and work item by PR comment and Chatter post respectively, e.g. GUS: [W-0000000: Work Name]()
- [x] Make sure that ACs are updated (if any gaps)
- [ ] **All acceptance criteria have been met**
    - [x] Developer
    - [ ] Code Reviewer
- [ ] Pull Request contains draft release notes
- [ ] Labels, help text, and customer facing messages are reviewed by Docs
- [ ] QE story level testing completed

### CODE REVIEW

**Comment by seanpat09** [comment]
[W-8901531 - OFM installer optionally installs OFM Experience](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B00000096pT0IAI/view)

File: cumulusci.yml
Comment (seanpat09) [inline cumulusci.yml:148]: @nicolersmith Can you review this installer message? This warning should appear in the installer if you don't have Digital Experiences enabled in your org. Without that setting enabled, the OFM Site Template installation would fail.

Screen Shot 2021-03-16 at 11 49 33 AM

File: cumulusci.yml
Comment (nicolersmith) [inline cumulusci.yml:145]:
```suggestion
              message: "Enable My Domain in your org prior to installing."
```

File: cumulusci.yml
Comment (nicolersmith) [inline cumulusci.yml:148]:
```suggestion
              message: "The Fundseeker Starter site template requires that Digital Experiences is enabled. Make sure to enable it in your org before installing the site template."
```

**Review by nicolersmith — APPROVED** [review]

File: cumulusci.yml
Comment (seanpat09) [reply cumulusci.yml:58]: @sella24 I wasn't sure how to set the options for the out of the box task in a when clause (i.e. `when: "not tasks.check_enable_networks_enabled()"`), so I created this task with the options set on it already. Is there a better way to do this?

Comment (sella24) [reply cumulusci.yml:58]: It makes sense that there should be a way to do this, but I also don't know how.
@scottwarren-sfdo  / @spelak-salesforce : Any idea if this is possible ?

File: cumulusci.yml
Comment (spelak-salesforce) [inline cumulusci.yml:102]: @seanpat09 , this is temporary?   Can we create a work item to change this to "install production"?

File: cumulusci.yml
Comment (spelak-salesforce) [reply cumulusci.yml:58]: We want to use this return value in a `when` clause, I don't think we have a mechanism to set options for a task inside a `when` clause.  Therefore (if this is accurate), we need to create our own implementation of this task where we can set the options we need.

Comment (sella24) [reply cumulusci.yml:58]: @jstvz : Is this possible ?

**Review by sella24 — APPROVED** [review]
I'll leave the cci to James.
Great job.

**Comment by sella24** [comment]
@seanpat09 : Can you also add instructions on how to test this in the PR ?
And a note that explains that we are using the beta for now and will change/update it later.

**Comment by seanpat09** [comment]
@jstvz can you create a staging installer for this PR so we can test the changes to the installer plan?

File: cumulusci.yml
Comment (jstvz) [reply cumulusci.yml:58]: This is the preferred approach. 

Now that there are multiple projects with this pattern we'll add a convenience task to CCI core.

File: cumulusci.yml
Comment (jstvz) [inline cumulusci.yml:102]: We can't install a beta in a customer org, so is this just for testing the step preflight?

**Review by jstvz — CHANGES_REQUESTED** [review]
Staging installer available at: https://metadeploy-stg.herokuapp.com/products/outbound-funds/cbbf024f

File: cumulusci.yml
Comment (seanpat09) [reply cumulusci.yml:102]: @jstvz yup! we can swap this out with `install_managed` before we merge this to `feature/231`

Comment (spelak-salesforce) [reply cumulusci.yml:102]: @seanpat09 , do we have a Work Item to swap this out?   If not, can we add one?

Comment (seanpat09) [reply cumulusci.yml:102]: @spelak-salesforce would that be covered by this work item? I was under the impression that this item wouldn't get closed until we swap it out

Comment (spelak-salesforce) [reply cumulusci.yml:102]: Yes as long as this does not merge until the `install_mangaged` task is referenced instead of the beta.   This will be easy to miss and merge without us realizing the installer doesn't work as expected.

File: cumulusci.yml
Comment (seanpat09) [inline cumulusci.yml:119]: @jstvz @sella24 Using this flow instead to optionally install OutboundFundsCommunity while also including the deploy_pre and deploy_post steps. Do either of you see any issues with taking this approach?

Comment (jstvz) [reply cumulusci.yml:119]: No issues with this approach.

Comment (seanpat09) [reply cumulusci.yml:119]: @jstvz Sounds good, do we need to spin up a new staging installer in order to reflect these new changes? If so, you can spin that up for us?

Comment (jstvz) [reply cumulusci.yml:119]: @seanpat09 https://metadeploy-stg.herokuapp.com/products/outbound-funds/65c7f649/install

File: cumulusci.yml
Comment (nicolersmith) [reply cumulusci.yml]: Hey @seanpat09, just noticed this today. Can you update this installer message to:

"After enabling Digital Experiences, create, configure, and then activate your new Experience Builder site so it's visible to your users."

**Comment by seanpat09** [comment]
This work item has been moved to never, closing PR and deleting branch

**Pull request closed without merging**
